frontend/src/components/Widgets/CustomEntitiesDropdown.tsx

```diff
@@ -324,41 +318,38 @@ export default function CustomEntitiesDropdown({
   };
 
   const handleInputChange = (event: any) => {
-    if (event.target.value !== undefined) {
-      onSearch(event.target.value, id).then(setAutocompleteOptions);
-    }
     setInputValue(event.target.value);
   };
 
   const isOptionSelected = (option: Option) => {
     return selectedOptions?.some((selected) => selected?.id === option?.id);
   };
-  
-  type tempOption = {sort: number} & Option
 
-  useEffect(() => {
-    const fetchData = async () => {
-      if (!isDropdownOpened) return;
-
-      setIsLoading(true);
-      try {
-        const options = await onSearch("", id);
-        const allOptions = [...selectedOptions, ...options];
-       
-        setAutocompleteOptions(selectType && selectType === 'from_entities' ? processFromEntitiesData(allOptions) : allOptions);
-      } catch (error) {
-        console.error("Error fetching data:", error);
-      } finally {
-        setIsLoading(false);
-      }
-    };
+  const fetchData = useCallback(async () => {
+    try {
+      const options = await onSearch(inputValue, id, selectedOptions);
+      const allOptions = [...selectedOptions, ...options];
+      const sortedOptions =
+        selectType && selectType === "from_entities"
```

Comment:
This check on selectType is not nice for a generic component. 
I would suggest to have a new prop instead. Something like 'postProcessOptions' or 'sortOptions' what do you think?